src/Behat/Testwork/Output/Printer/JUnitOutputPrinter.php

```diff
@@ -57,6 +57,10 @@ public function __construct(FilesystemOutputFactory $outputFactory)
      */
     public function createNewFile($name, array $testsuitesAttributes = array())
     {
+        // This requires the DOM extension to be enabled.
+        if (!extension_loaded('dom')) {
+            throw new \Exception('The PHP DOM extension is required to generate JUnit reports.');
```

Comment:
I'd throw something more specific, like `RuntimeException`.

Reply:
There are also some more Printer specific exceptions in this package: https://github.com/Behat/Behat/tree/master/src/Behat/Testwork/Output/Exception

I think the intention might've been for Printer implementations to throw exceptions implementing the PrinterException.